lincheck/src/main/java/org/jetbrains/kotlinx/lincheck/verifier/LTS.kt

```diff
@@ -84,313 +91,377 @@ abstract class AbstractLTSVerifier<STATE>(val scenario: ExecutionScenario, val t
  */
 class LTS(
     private val testClass: Class<*>,
-    private val relaxationFactor: Int
+    private val isQuantitativelyRelaxed: Boolean,
+    private val relaxationFactor: Int,
+    private val maxTicket: Int
 ) {
-    // LTS states cache
-    // Store the state and tickets assigned to operations leading to this state in order
-    // to remap tickets assigned to the new sequence of operations leading to this state
-    private val states: MutableMap<StateKey, StateInfo> = HashMap()
 
-    // Stores tickets of resumed operations with corresponding resumption information
-    private val resumedTicketsWithResults = mutableMapOf<Int, ResumedResult>()
+    private val stateInfos = HashMap<StateInfo, StateInfo>()
 
     /**
      * Initial LTS state
      */
-    public val initialState = getOrCreateState(
-        instanceState = Utils.createTestInstance(testClass, relaxationFactor),
-        suspendedOperations = mutableListOf(),
-        seqToCreate = emptyList()
-    ).first.state
+    public val initialState: State = createInitialState()
 
-    inner class State(var seqToCreate: List<Operation>, val costCounter: Any? = null) {
+    inner class State(var seqToCreate: List<Operation>, private val costCounter: Any? = null) {
         // Contains transitions by actors with threads available from this state
-        private val transitions = mutableMapOf<Operation, List<TransitionInfo>>()
-
-        fun next(actor: Actor, expectedResult: Result, requestTicket: Int?): List<TransitionInfo> {
-            val operation = Operation(actor, expectedResult, requestTicket, requestTicket == null)
-            return transitions.computeIfAbsent(operation) {
-                val (instanceState, suspendedOperations) = createState()
-                val res = operation.invoke(instanceState, suspendedOperations)
-                computeNextTransitions(instanceState, suspendedOperations, operation, res)
+        private val transitions = mutableMapOf<Pair<Actor, Int>, TransitionInfo?>()
```

Comment:
We have to distinguish between transitions by `request` and `follow-up` parts of the same actor. 
I'll fix it by creating a wrapping object anyway :frowning_woman: